### PR TITLE
feat: support the new workspace folder structure

### PR DESCRIPTION
**Pull request by HarikrishnanBalagopal**
Signed-off-by: Harikrishnan Balagopal <harikrishmenon@gmail.com>

### CODE REVIEW

File: internal/application/filesystem.go [hunk 1]
```diff
@@ -519,112 +550,148 @@ func (a *FileSystem) DeletePlan(appName string) error {
...
+			if err := CopyDir(custPath, currentRunCustDir); err != nil {
+				log.Errorf("failed to copy the customizations directory for the current run. Error: %q", err)
+				return err
+			}
+		} else {
+			log.Debugf("the customizations directory has already been copied for the current run")
+		}
+		cmdArgs = append(cmdArgs, "--customizations", currentRunCustDir)
+	}
+	log.Infof("transform cmdArgs!!!!!!!!!!!!!!!!!!!!!!!!!!: %+v", cmdArgs)
```
Comment (ashokponkumar) [inline internal/application/filesystem.go:664]: Do we want to remove this comment?

File: internal/move2kubeapi/move2kubeapi.go [hunk 2]
```diff
@@ -135,20 +134,25 @@ func getAssetsList(w http.ResponseWriter, r *http.Request, isCustomization bool)
...
+		log.Errorf("failed to marhsal the assets/customizations list %+v to json. Error: %q", assets, err)
 	}
 }
 
 func getAsset(w http.ResponseWriter, r *http.Request, isCustomization bool) {
 	name := mux.Vars(r)["name"]
-	asset := mux.Vars(r)["asset"]
+	asset := ""
+	if isCustomization {
+		asset = mux.Vars(r)["customization"]
```
Comment (ashokponkumar) [inline internal/move2kubeapi/move2kubeapi.go:152]: Do we want to keep the flag name same as in command line? "customization**s**"

File: internal/move2kubeapi/move2kubeapi.go [hunk 3]
```diff
@@ -159,13 +163,18 @@ func getAsset(w http.ResponseWriter, r *http.Request, isCustomization bool) {
...
+		log.Errorf("failed to send the asset/customization to the client. Error: %q", err)
 	}
 }
 
 func deleteAsset(w http.ResponseWriter, r *http.Request, isCustomization bool) {
 	name := mux.Vars(r)["name"]
-	asset := mux.Vars(r)["asset"]
+	asset := ""
+	if isCustomization {
+		asset = mux.Vars(r)["customization"]
```
Comment (ashokponkumar) [inline internal/move2kubeapi/move2kubeapi.go:174]: Since we are using the string in multiple places, do we want to change it to a const?

**Review by ashokponkumar — CHANGES_REQUESTED** [review]
Nice work! Few comments.

File: internal/application/filesystem.go [hunk 4]
```diff
@@ -53,14 +53,14 @@ const (
...
-	timestampRegex                          = `time="\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}Z"`
 	loglevelRegex                           = `level=([a-z]+) `
 	newfilesMetadataFileName                = "newfiles.txt"
 	defaultDirectoryPermissions os.FileMode = 0777
 	defaultFilePermissions      os.FileMode = 0777
 )
 
 var (
+	timestampRegex   = regexp.MustCompile(`time="\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}Z"`)
 	validArchiveExts = []string{".zip", ".tar", ".tgz", ".gz"}
```
Comment (ashokponkumar) [inline internal/application/filesystem.go:64]: Will this support .tar.gz and .tar.xz? I assume at least .tar.gz is quite common.

File: internal/application/filesystem.go (hunk 1 above)
Comment (HarikrishnanBalagopal) [reply internal/application/filesystem.go]: done

**Review by ashokponkumar — CHANGES_REQUESTED** [review]

File: internal/move2kubeapi/move2kubeapi.go (hunk 2 above)
Comment (HarikrishnanBalagopal) [reply internal/move2kubeapi/move2kubeapi.go:152]: it is the same, if you look at the url endpoints I have tried to keep it analogous to the `assets`
https://github.com/konveyor/move2kube-api/blob/main/internal/move2kubeapi/move2kubeapi.go#L388-L396

File: internal/move2kubeapi/move2kubeapi.go (hunk 3 above)
Comment (HarikrishnanBalagopal) [reply internal/move2kubeapi/move2kubeapi.go:174]: hmm, there are lot of strings like `asset`, `name`, etc. that are also hardcoded. Should we change everything to be const?

File: internal/application/filesystem.go (hunk 4 above)
Comment (HarikrishnanBalagopal) [reply internal/application/filesystem.go:64]: I didn't change the regex. I just moved the regex as is to a `var` block so it can be reused.

Comment (HarikrishnanBalagopal) [reply internal/application/filesystem.go:64]: That regex is not dealing with any archive file extensions.
It's being used to replace the timestamp in log lines with the app name.

File: internal/move2kubeapi/move2kubeapi.go (hunk 3 above)
Comment (ashokponkumar) [reply internal/move2kubeapi/move2kubeapi.go:174]: if there is a name referenced in more than one place, and it needs to be consistent, then yes.

File: internal/application/filesystem.go (hunk 4 above)
Comment (ashokponkumar) [reply internal/application/filesystem.go:64]: Okay, we can extend the archive formats we support some other time then.

**Review by ashokponkumar — APPROVED** [review]

**Pull request merged**